actions/hadoop-upgrade

```diff
@@ -0,0 +1,253 @@
+        exit 1
+fi
+
+if ! [ "${forceupgrade}" == "True" ] ; then
+        query_upgrade
+fi
+
+if [ "${postupgrade}" == "downgrade" ] ; then
+        if [ "${standalone}" == "False" ] ; then
+                query_upgrade
+        fi
+        if [ -d /usr/lib/hadoop-${new_hadoop_ver} ] ; then
+                rm /usr/lib/hadoop
+                ln -s /usr/lib/hadoop-${new_hadoop_ver} /usr/lib/hadoop
+                if [ -d /usr/lib/hadoop-${current_hadoop_ver}/logs ] ; then
+                        mv /usr/lib/hadoop-${current_hadoop_ver}/logs /usr/lib/hadoop/
+                fi
+                action-set newhadoop.downgrade="successfully downgraded, ready for finalize"
+                status-set active "Ready - downgrade to ${new_hadoop_ver} complete, ready for finalize"
+                init_procs stop 
+                init_procs start
+                exit 0
+        else
+                if [ "${forceupgrade}" == "False" ] ; then
+                        action-set newhadoop.downgrade="previous version not found, aborting"
+                        action-fail "previous version not found, cannot downgrade - try forcing"
+                        status-set active "previous version not found, aborting"
+                        exit 1
+                fi
+        fi
+fi
+
+if [ "${postupgrade}" == "disabled_rollback" ] ; then
+        if is-leader && [ ${is_namenode} == "true" ] ; then
+                su hdfs -c "stop-dfs.sh"
+        fi
+        if [ -d /usr/lib/hadoop-${new_hadoop_ver} ] ; then
+                rm /usr/lib/hadoop
+                ln -s /usr/lib/hadoop-${new_hadoop_ver} /usr/lib/hadoop
+                if [ -d /usr/lib/hadoop-${current_hadoop_ver}/logs ] ; then
+                        mv /usr/lib/hadoop-${current_hadoop_ver}/logs /usr/lib/hadoop/
+                fi
+                action-set newhadoop.rollback="successfully rolled back"
+                status-set active "Ready - rollback to ${new_hadoop_ver} complete - ready for rollback_finalize"
+                exit 0
+        else
+                action-set newhadoop.rollback="previous version not found, unpacking..."
+                status-set active "previous version not found, unpacking..."
+        fi
+fi
+
+if [ "${postupgrade}" == "disabled_rollback_finalize" ] ; then
+        is_namenode=`chlp unitdata get is.namenode`
+        if [ ${is_namenode} == "true" ] ; then
+                if is-leader ; then
+                        su hdfs -c "hadoop-daemon.sh start namenode -rollingUpgrade rollback"
+                        exit 0 
+                else
+                        sleep 60
+                        su hdfs -c "hdfs namenode -bootstrapStandby"
+                        su hdfs -c "start-dfs.sh -rollback"
+                        exit 0
+                fi
+        fi
+fi       
+
+
+
+status-set maintenance "Fetching hadoop-${new_hadoop_ver}-${cpu_arch}"
+juju-resources fetch hadoop-${new_hadoop_ver}-${cpu_arch}
```

Comment:
I guess the action could parse the resources.yaml for the available versions and if the correct one isnt specified it returns a list of what is supported. The output could also be added as an output to an optional param such as listversions=true